### ISSUE TITLE
set function does not seem to work

### ISSUE THREAD

**Issue by ruben-dossantos** [shineum/cordova-plugin-pm80-scanner#1]
When i use the function scan in my app it turns on the laser and reads the barcodes. However when i use the set function, it beeps but it does not call the success or error callback function.

I have something like

    var successCallback = function(s) {
            $scope.my_barcode = s;
    };
    var errorCallback = function(error) {
            $scope.f.putError("ERRO!")
    };
    cordova.plugins.pm80scanner.beep(true);
    cordova.plugins.pm80scanner.set(successCallback, errorCallback);

Am I missing something?

**Comment by shineum** [comment]
Can you try this code? 
Instead of using delegates, put callback functions directly as parameters.


```
cordova.plugins.pm80scanner.beep(true);
cordova.plugins.pm80scanner.set( 
   function(s) {
        $scope.my_barcode = s;
   },
   function(error) {
        $scope.f.putError("ERRO!");
   }
);
```

**Comment by ruben-dossantos** [comment]
It works the first time.. Maybe the problem is that I am using Angular.

I went and done something like this

    laserFn = function(){
                cordova.plugins.pm80scanner.scan(function(s) {
                        $scope.my_barcode = s
                    }, function(error){...});
            }

This allows me to press a button on the frontend and scan. The only setback is that i can't activate it by pressing the side buttons. Is there away to register that click through angular/javascript?

**Comment by larkonesvk** [comment]
Hi,
i can agree with xumbino. Callback is fired only once. Next time when i scan it beeps but callback is not fired.

`           
 cordova.plugins.pm80scanner.set(function (result) {
                alert(result);
            }, function (error) {
                alert("Error: "+ error);
            });
`

alert is fired for first time and then never again.

**Comment by shineum** [comment]
I don't think It is the Angular (or Ionic) issue.

And "set" function is designed to receive scanner callback just once. 
However, if you want to receive scanner callback repeatedly, just set it again at the end of your callback code.

Here is the example code.


```

function addScannerListner() {
    cordova.plugins.pm80scanner.set(
        function(result) {
            successCallback(result);
            addScannerListener(); 
        }, 
        function(error) {
            errorCallback(error);
            addScannerListener(); 
        }
    );
}

```

**Comment by ruben-dossantos** [comment]
Sorry, I tried that as you suggested and it only works the first time.

    $scope.n = 0;
    function successCallback(s){
            $scope.my_barcode = s;
            $scope.scanner();
        }
    function errorCallback(error){
            console.log(error);
            $scope.scanner();
        }

    $scope.scanner = function(){
            $scope.n = $scope.n + 1;
            $scope.state = cordova.plugins.pm80scanner.set(
                function(s){
                    successCallback(s);
                },
                function(error){
                    errorCallback(error);
                }
            )
        };
    $scope.scanner();

My "n" variable "starts" at 1, and when I scan it goes to 2, however it does not scan again neither "n" changes to 3.

Even though, if pm80scanner.set is working as intended, you should add a way to natively scan continuously, as I think it would be a great addition to this plugin.

**Comment by larkonesvk** [comment]
@xumbino as shineum said you have to call cordova.plugins.pm80scanner.set inside your successCallback. I didnt try it yet.
But i think its not good pattern to call set function inside successCallback. When i scan 10 barcodes in 30 sec i would always have to set callbacks. I think it would case worse performance. So i agree with xumbino to add continously scan. (datawedge plugin this supports)

**Comment by ruben-dossantos** [comment]
@larkonesvk yes, i tried both ways but neither worked..

**Comment by shineum** [comment]
@xumbino @larkonesvk 
Okay guys, I'm gonna update this plugin to support native continuous scan function soon.
And if you have any other improvements, let me know. 
Thank you!

**Comment by larkonesvk** [comment]
Great. It would be nice to have function for unregister these callbacks. So when barcode is scanned callback is not fired.

Thank you too

**Comment by larkonesvk** [comment]
@xumbino try this

```
addScanListener: function(){
            cordova.plugins.pm80scanner.set(
                    function (result) {
                        alert(result);
                        addScannerListener();
                    },
                    function (error) {
                        alert("error");
                        addScannerListener();
                    }
            ); 
        }
```

It works very well for me. Until shineum add native support for continuous scan its good enough.

**Comment by shineum** [comment]
@larkonesvk @xumbino 
I just released v0.0.5 to support changing trigger mode.
And updated to v0.0.6, because of a typo on description.
Thanks!

**Comment by ruben-dossantos** [comment]
Awesome, I'll try it right away and get back at you. ;)

**Comment by ruben-dossantos** [comment]
Works for me :+1:

**Comment by shineum** [comment]
Okay, then I'll close this issue.
Thank you!
